### ISSUE TITLE
pact-file URL with file: scheme is unable to be loaded due to cast exception (3.5.2 onwards)

### ISSUE THREAD

**Issue by delitescere** [DiUS/pact-jvm#733]
Something happened in 3.5.2 where specifying a pact-file with a `file:` schema fails.

In my lein project I specify a consumer with `:pact-file "file:test/pact/default-consumer.json"`

Up to version 3.5.1 this works fine. Any version above that fails with:

```sh
org.codehaus.groovy.runtime.typehandling.GroovyCastException: Cannot cast object 'file:test/pact/default-consumer.json' with class 'java.net.URL' to class 'java.io.File'
 at org.codehaus.groovy.runtime.typehandling.DefaultTypeTransformation.continueCastOnSAM (DefaultTypeTransformation.java:405)
    org.codehaus.groovy.runtime.typehandling.DefaultTypeTransformation.continueCastOnNumber (DefaultTypeTransformation.java:319)
    org.codehaus.groovy.runtime.typehandling.DefaultTypeTransformation.castToType (DefaultTypeTransformation.java:232)
    org.codehaus.groovy.runtime.DefaultGroovyMethods.asType (DefaultGroovyMethods.java:15537)
    org.codehaus.groovy.runtime.dgm$51.doMethodInvoke (:-1)
    org.codehaus.groovy.vmplugin.v7.IndyInterface.selectMethod (IndyInterface.java:232)
    au.com.dius.pact.provider.ConsumerInfo.setPactFile (ConsumerInfo.groovy:40)
```

**Comment by uglyog** [comment]
Runtime exceptions are the bane of dynamic languages.

**Comment by uglyog** [comment]
Released 3.5.20 with this fix
